src/scitokens_internal.cpp

```diff
@@ -84,6 +88,20 @@ SimpleCurlGet::GetStatus SimpleCurlGet::perform_start(const std::string &url) {
     return perform_continue();
 }
 
+std::string SimpleCurlGet::get_url() const {
+    if (!m_curl) {
+        return "";
+    }
+
+    char *url = nullptr;
+    auto rv = curl_easy_getinfo(m_curl.get(), CURLINFO_EFFECTIVE_URL, &url);
```

Comment:
Do we need to free the url memory?

Reply:
It appears no. Check the last sentence of `Description` from https://linux.die.net/man/3/curl_easy_getinfo